core/src/test/java/com/netflix/iceberg/TestReplaceFiles.java

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.iceberg;
+
+import com.google.common.collect.Iterators;
+import com.google.common.collect.Lists;
+import com.netflix.iceberg.exceptions.CommitFailedException;
+import com.netflix.iceberg.exceptions.ValidationException;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.internal.util.collections.Sets;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.Iterator;
+
+import static com.google.common.collect.Iterators.concat;
+import static com.netflix.iceberg.ManifestEntry.Status.*;
+
+public class TestReplaceFiles extends TableTestBase {
+
+  @Test(expected = ValidationException.class)
+  public void testEmptyTable() {
+    Assert.assertEquals("Table should start empty", 0, listMetadataFiles("avro").size());
+
+    TableMetadata base = readMetadata();
+    Assert.assertNull("Should not have a current snapshot", base.currentSnapshot());
+
+    table.newRewrite()
+        .replaceFiles(Sets.newSet(FILE_A), Sets.newSet(FILE_B))
+        .apply();
+  }
+
+  @Test(expected = ValidationException.class)
+  public void testAddOnly() {
+    Assert.assertEquals("Table should start empty", 0, listMetadataFiles("avro").size());
+
+    table.newRewrite()
+            .replaceFiles(Sets.newSet(FILE_A), Collections.emptySet())
+            .apply();
+  }
+
+  @Test(expected = ValidationException.class)
+  public void testDeleteOnly() {
+    Assert.assertEquals("Table should start empty", 0, listMetadataFiles("avro").size());
+
+    table.newRewrite()
+            .replaceFiles(Collections.emptySet(), Sets.newSet(FILE_A))
+            .apply();
+  }
+
+  @Test
+  public void testAddAndDelete() {
```

Comment:
This should also verify that there are no unexpected manifests with `listMetadataFiles`.

Reply:
Added the check.